### DISCUSSION TITLE
How do I run the script hourly without my terminal open?

### DISCUSSION THREAD

**Discussion by dannyvfilms** [meisnate12/Plex-Meta-Manager#458] [Q&A]
I went on a bit of a journey to figure out how to update my Trakt Trending collection hourly, and I still don't think I've figured it out. First I thought I needed to had `schedule: hourly` in my Movies.yml file, but that didn't change anything. So based on the wording I wondered if I needed to indicate which hours it would run:

`  Trending Right Now:
    trakt_trending: 30
    sort_title: "!B Trending Right Now"
    sync_mode: sync
    radarr_add: false
    collection_order: custom
    schedule: hourly(1), hourly(2), hourly(3), hourly(4), hourly(5), hourly(6), hourly(7), hourly(8), hourly(9), hourly(10), hourly(11), hourly(12), hourly(13), hourly(14), hourly(15), hourly(16), hourly(17), hourly(18), hourly(19), hourly(20), hourly(21), hourly(22), hourly(23)
    visible_home: true
    visible_shared: true`

That didn't seem to fix it either, so I started playing with the main run command. I thought if I told it in run command to start every hour that would work: `python plex_meta_manager.py --config /config/config.yml --time 00:00,01:00,02:00,03:00,04:00,05:00,06:00,07:00,08:00,09:00,10:00,11:00,12:00,13:00,14:00,15:00,16:00,17:00,18:00,19:00,20:00,21:00,22:00,23:00`

That also failed, since it didn't know what time it was or know that I wanted it to run in the next hour: `| Current Time: 07:05 | 19 Hours and 55 Minutes until the next run at 03:00 | Runs: 3:00`

Once I solve all of this, I don't think the script is running offscreen. Only when I open a console from Unraid and put in a command. What's the proper workflow for running this offscreen every hour? I know the wiki is extensive, and I've missed things before, but I can't seem to get my head around what seemed to be reasonable things to try.

**Comment by meisnate12** [comment]
this `python plex_meta_manager.py --config /config/config.yml --time 00:00,01:00,02:00,03:00,04:00,05:00,06:00,07:00,08:00,09:00,10:00,11:00,12:00,13:00,14:00,15:00,16:00,17:00,18:00,19:00,20:00,21:00,22:00,23:00` 

might need `"` around the times like this 

`python plex_meta_manager.py --config /config/config.yml --time "00:00,01:00,02:00,03:00,04:00,05:00,06:00,07:00,08:00,09:00,10:00,11:00,12:00,13:00,14:00,15:00,16:00,17:00,18:00,19:00,20:00,21:00,22:00,23:00"`

because when run like that you shouldn't see this `| Current Time: 07:05 | 19 Hours and 55 Minutes until the next run at 03:00 | Runs: 3:00`
that last runs section should be all your scheduled runs. are you using the most update version of PMM?

**Reply by dannyvfilms** [reply]
I’ll try the “ when I get home. Didn’t realize that was the syntax. I should be on the latest develop from an issue you helped me with about two days ago.

**Reply by dannyvfilms** [reply]
I think the Docker variable `PMM_TIME` might have gotten in the way. It defaults to `03:00`, so I changed it to `00:00,01:00,02:00,03:00,04:00,05:00,06:00,07:00,08:00,09:00,10:00,11:00,12:00,13:00,14:00,15:00,16:00,17:00,18:00,19:00,20:00,21:00,22:00,23:00` and now the `| Runs: ...` portion at the end shows all of the scheduled times. I *think* it should be good now.

**Reply by meisnate12** [reply]
Yes Environmental Variables are used over the command-line arguments so this should work
